profile/middle_weight_benchmark.rb

```diff
@@ -0,0 +1,233 @@
+#
+# @return [ [Array<Integer>, Double, String] ] An array of benchmark wall clock time results, the size of the dataset in MB, test label
+#
+# @since 2.2.1
+def small_doc_bulk_insert(database, collection, reps)
+  small_doc_data = BenchmarkHelper.load_array_from_file(SMALL_DOC)
+  data_file_size = File.size(SMALL_DOC)
+
+  tms_results = []
+  reps.times do
+    database.drop
+    tms_results << Benchmark.bm do |bm|
+      bm.report('Middleweight::Small doc bulk insert') do
+        collection.insert_many(small_doc_data, ordered: true)
+      end
+    end.first
+  end
+
+  # Get the real times (wall clock times) from the Benchmark::Tms objects
+  return tms_results.map { |result| result.real }, data_file_size/1000000.0, "Small doc bulk insert"
+end
+
+
+# Large doc bulk insert
+#
+# - Drop the database.
+# - Load LARGE_DOC dataset
+#
+# Measure: Do an ordered bulk insert of all 1,000 documents.
+#          DO NOT manually add an _id field
+#
+# @param [ Mongo::Database ] database MongoDB Database.
+# @param [ Mongo::Collection ] collection The collection.
+# @param [ Integer ] reps Number of repetitions of the benchmark to run.
+#
+# @return [ [Array<Integer>, Double, String] ] An array of benchmark wall clock time results, the size of the dataset in MB, test label
+#
+# @since 2.2.1
+def large_doc_bulk_insert(database, collection, reps)
+  large_doc_data = BenchmarkHelper.load_array_from_file(LARGE_DOC)
+  data_file_size = File.size(LARGE_DOC)
+
+  tms_results = []
+  reps.times do
+    database.drop
+    tms_results = Benchmark.bm do |bm|
+      bm.report('Middleweight::Large doc bulk insert') do
+        collection.insert_many(large_doc_data, ordered: true)
+      end
+    end
+  end
+
+  # Get the real times (wall clock times) from the Benchmark::Tms objects
+  return tms_results.map { |result| result.real }, data_file_size/1000000.0, "Large doc bulk insert"
+end
+
+
+##
+# GridFS upload
+#
+# - Drop the database.
+# - Load the GRIDFS_LARGE file as a string.
+#
+# Measure: Upload the GRIDFS_LARGE data as a GridFS file 100 times.  Each time, use a
+#          different filename, following the sprintf pattern "gridfstest%03d" with integers
+#          from 1 to 100.
+#
+# @param [ Mongo::Database ] database MongoDB Database.
+# @param [ Integer ] reps Number of repetitions of the benchmark to run.
+#
+# @return [ [Array<Integer>, Double, String] ] An array of benchmark wall clock time results, the size of the dataset in MB, test label
+#
+# @since 2.2.1
+def gridfs_upload(database, reps)
+  gridfs_large_data_string = BenchmarkHelper.load_string_from_file(GRIDFS_LARGE)
+  data_file_size = File.size(GRIDFS_LARGE)
+
+  tms_results = []
+  reps.times do
+    database.drop
+    tms_results << Benchmark.bm do |bm|
+      bm.report('Middleweight::GridFS upload') do
+        100.times do |i|
+          database.fs.insert_one(
+              Mongo::Grid::File.new(gridfs_large_data_string, :filename => "gridfstest%03d" % (i+1))
```

Comment:
I added an array of filenames. But wouldn't uploading from a stream take more time since it's reading the data from a file (I/O operation) and then sending the data to the DB? This way the data is already stored in a string.